src/Process/Signal.php

```diff
@@ -13,31 +14,40 @@ class Signal implements SignalInterface
     use Defensive\AwareTrait;
     use Process\Signal\Information\Factory\AwareTrait;
     use Process\Pool\Logger\AwareTrait;
+
+    protected const HANDLE_SIGNAL = 'handleSignal';
+
     protected $signalHandlers = [];
     protected $bufferedSignals = [];
+    protected $canBufferSignals = true;
 
     public function addSignalHandler(int $signalNumber, HandlerInterface $signalHandler): SignalInterface
     {
         pcntl_async_signals(true);
         $this->signalHandlers[$signalNumber] = $signalHandler;
-        pcntl_signal($signalNumber, [$this, 'handleSignal']);
+        pcntl_signal($signalNumber, [$this, self::HANDLE_SIGNAL]);
 
         return $this;
     }
 
     public function processBufferedSignals(): SignalInterface
     {
         foreach ($this->bufferedSignals as $position => $information) {
-            unset($this->bufferedSignals[$position]);
-            $this->processSignalInformation($information);
+            try {
+                $this->processSignalInformation($information);
+                unset($this->bufferedSignals[$position]);
+            } catch (\Throwable $throwable) {
+                unset($this->bufferedSignals[$position]);
```

Comment:
I'm not sure I understand this, do we want to remove that signal from the buffer if it couldn't be processed?

If that's the case then my previous comment about this doesn't really apply

Reply:
> do we want to remove that signal from the buffer if it couldn't be processed?

yes

If that's the case then my previous comment about this doesn't really apply

ok